server/bleep/src/webserver/answer.rs

```diff
@@ -249,54 +244,58 @@ pub async fn handle(
         }
     };
 
-    let explain_prompt = answer_api_client.build_explain_prompt(&processed_snippet);
-    let snippet_explanation = answer_api_client
-        .explain_snippet(&explain_prompt)
-        .await
-        .map_err(|e| {
-            sentry::capture_message(
-                format!("answer-api failed to respond: {e}").as_str(),
-                sentry::Level::Error,
-            );
-            super::error(ErrorKind::UpstreamService, e.to_string())
-        })?
-        .text()
-        .await
-        .map_err(super::internal_error)?;
-
     // reorder snippets
     snippets.swap(relevant_snippet_index, 0);
 
     let query_id = uuid::Uuid::new_v4();
-    app.track_query(QueryEvent {
-        user_id: params.user_id.clone(),
-        query_id,
-        query: params.q.clone(),
-        semantic_results: all_snippets,
-        filtered_semantic_results: snippets.clone(),
-        select_prompt,
-        relevant_snippet_index,
-        explain_prompt,
-        explanation: snippet_explanation.clone(),
-        overlap_strategy: semantic.overlap_strategy(),
-    });
 
     // answering snippet is always at index 0
     let answer_path = snippets.get(0).unwrap().relative_path.to_string();
 
-    Ok::<_, Json<super::Response<'static>>>(Json(super::Response::Answer(AnswerResponse {
-        snippets,
-        query_id,
-        user_id: params.user_id,
-        selection: api::Response {
-            answer: snippet_explanation,
+    let initial_event = Event::default()
+        .json_data(super::Response::Answer(AnswerResponse {
```

Comment:
can we implement `Into<super::Response> for AnswerResponse`, like we do for all these response types? like how `HoverableResponse` and `TokenInfoResponse` convert to `super::Response`. we could every macro-ify this impl in `webserver.rs`.